src/overlays/actors/ovl_En_Heishi/z_en_heishi.c

```diff
@@ -30,33 +43,148 @@ const ActorInit En_Heishi_InitVars = {
+        OCELEM_ON,
+    },
     { 20, 60, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80BE9450;
-
-extern UNK_TYPE D_06003BFC;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/EnHeishi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/EnHeishi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/func_80BE90BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/func_80BE9148.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/func_80BE91DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/func_80BE9214.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/EnHeishi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/func_80BE9380.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Heishi/EnHeishi_Draw.s")
+void EnHeishi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHeishi* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 25.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gSoldierSkeleton, &gSoldierWave, this->jointTable,
+                       this->morphTable, HEISHI_LIMB_MAX);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->paramCopy = this->actor.params;
+    this->yawTowardsPlayer = this->actor.world.rot.y;
+
+    if (!this->paramCopy) {
+        this->shouldSetHeadRotation = 1;
+        if (!(gSaveContext.weekEventReg[63] & ACTOR_FLAG_80) && ((gSaveContext.day != 3) || !gSaveContext.isNight)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        this->colliderCylinder.dim.radius = 30;
+        this->colliderCylinder.dim.height = 60;
+        this->colliderCylinder.dim.yShift = 0;
+        if ((gSaveContext.weekEventReg[63] & ACTOR_FLAG_80) || ((gSaveContext.day == 3) && gSaveContext.isNight)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    this->actor.flags |= ACTOR_FLAG_8000000;
+    EnHeishi_SetupIdle(this);
+}
+
+void EnHeishi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHeishi* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnHeishi_ChangeAnimation(EnHeishi* this, s32 animIndex) {
+    static AnimationHeader* sAnimations[] = { &gSoldierStandHandOnHip, &gSoldierCheerWithSpear, &gSoldierWave,
+                                              &gSoldierSitAndReach, &gSoldierStandUp };
+    static u8 sAnimModes[] = { ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP,
+                               ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP,
+                               ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP };
```

Comment:
```suggestion
    static AnimationHeader* sAnimations[] = { &gSoldierStandHandOnHip, &gSoldierCheerWithSpear, &gSoldierWave,
                                              &gSoldierSitAndReach, &gSoldierStandUp, };
    static u8 sAnimModes[] = { ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP,
                               ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP,
                               ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, };
```
Would probably format better with trailing commas.

Reply:
What does the trailing comma do when the format.sh script runs?

Reply:
The best resource that I could find about it is `http://www.devans.xyz/2017/02/22/clang-format.html`. The main thing is it splits individual elements into their own line. Note also how with no trailing comma, everything gets aligned after the array declaration, i.e. to the right, while with a trailing comma everything starts and gets aligned below the array declaration.